haystack/__init__.py

```diff
@@ -23,6 +23,7 @@
     "default_to_dict",
     "DeserializationError",
     "ComponentError",
+    "AsyncPipeline",
```

Comment:
(nit) suggestion: keeping these imports ordered alphabetically helps locate something as the list grows

```python
__all__ = [
    "Answer",
    "AsyncPipeline",
    "ComponentError",
    "DeserializationError",
    "Document",
    "ExtractedAnswer",
    "GeneratedAnswer",
    "Pipeline",
    "PredefinedPipeline",
    "component",
    "default_from_dict",
    "default_to_dict",
]
```